### PR TITLE
[notifications][android] add clarifying comments to code

### PR DESCRIPTION
**Pull request by vonovak**
# Why

I find there's a lot going on and having some "checkpoints" / comments helps me get around. It should help others too. 

# How

Added some explanatory comments, and some TODO comments to be handled later. This PR doesn't change any runtime behavior.

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/model/NotificationAction.java [hunk 1]
```diff
@@ -7,6 +7,9 @@
 
 /**
  * A class representing a single notification action button.
+ *
+ * TODO vonovak: no need to implement serializable, parcelable is enough for storing
```
Comment (douglowder) [inline packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/model/NotificationAction.java:11]: I had thought that actions were serialized and sent to JS as part of events, so maybe Serializable is needed here?

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/model/NotificationCategory.java [hunk 2]
```diff
@@ -9,6 +9,8 @@
 
 /**
  * A class representing a collection of notification actions.
+ *
+ * TODO vonovak: no need to implement serializable, parcelable is enough for storing
```
Comment (douglowder) [inline packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/model/NotificationCategory.java:13]: I had thought that categories were serialized and sent to JS as part of events, so maybe Serializable is needed here?

**Review by douglowder — APPROVED** [review]
A couple of questions, but looks good otherwise!

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/model/NotificationAction.java (hunk 1 above)
Comment (vonovak) [reply packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/model/NotificationAction.java:11]: Hello, the response to a button press is sent to JS via [here](https://github.com/expo/expo/blob/6845f2657d4deece98f925b7165f49944f8016d7/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/NotificationSerializer.java#L42-L42)

and the actions are passed from JS to native via [here](https://github.com/expo/expo/blob/6845f2657d4deece98f925b7165f49944f8016d7/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/categories/ExpoNotificationCategoriesModule.kt#L98-L98)

I might be wrong but I'm almost certain there's no need to implement the Serializable interface.

**Pull request merged**